src/app/common/components/CertificateUpload.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { render } from 'react-dom';
+import {
+  Button,
+  InputGroup,
+  TextInput
+} from '@patternfly/react-core';
+
+interface ICertificateUploadProps {
+  isDisabled: boolean;
+  onBlur: any;
+  onChange: any;
+  onInput: any;
+  name: string;
+}
+
+interface ICertificateUploadState {
+  display: string;
+}
+
+class CertificateUpload extends React.Component<ICertificateUploadProps, ICertificateUploadState> {
+
+  private fileInput = React.createRef<HTMLInputElement>();
+
+  constructor(props) {
+    super(props);
+    this.clickFileInput = this.clickFileInput.bind(this);
+    this.state = {
+      display: '',
+    };
+  }
+
+  clickFileInput() {
+    // redirect a click from any other part of the
+    // component to the hidden file input in order
+    // to open the browse dialog
+    const fileInput = this.fileInput.current;
+    if (fileInput) {
+      fileInput.click();
+    }
+  }
+
+  readFile = fileHandle => {
+    // read the uploaded certificate and b64 encode it, then
+    // save it to the bound form field.
+    const reader = new FileReader();
+    reader.onload = (event) => {
+      const newValue = btoa(reader.result as string);
+      this.setState({ display: fileHandle.name });
+      this.props.onChange(this.props.name, newValue);
+    };
+    reader.readAsBinaryString(fileHandle);
+  };
```

Comment:
replace with: 

`const readFile = fileHandle => {
    // read the uploaded certificate and b64 encode it, then
    // save it to the bound form field.
    const reader = new FileReader();
    reader.onload = (event) => {
      const newValue = btoa(reader.result as string);

      setDisplay(fileHandle.name);
      setFieldValue(name, newValue);
    };
    reader.readAsBinaryString(fileHandle);
  };